common/djangoapps/student/apps.py

```diff
@@ -23,3 +23,13 @@ def ready(self):
         from django.contrib.auth.models import User
         from .signals.receivers import on_user_updated
         pre_save.connect(on_user_updated, sender=User)
+
+        # The django-simple-history model on CourseEnrollment creates performance
+        # problems in testing, we mock it here so that the mock impacts all tests.
+        from django.contrib.auth import settings
+
+        if hasattr(settings, 'TEST_ROOT'):
```

Comment:
I could not find a better / edX standard way of determining that we are running in a test context. Open to suggestions here.

Reply:
I'd like to see Test Engineering give their blessing on this point - a standard way of determining if code is running in the context of testing vs. the application itself. The rest of the PR: 👍

Reply:
Rather than doing this here, why not just monkey patch the import in the test files?

Reply:
@doctoryes TestEng had a suggested path that I'm going to investigate now, similar to the suggestion that we do it through Paver.

@feanil There are several hundred places, perhaps as many as a thousand, that would need to be mocked, and no good way to extend this to new tests as they get written. I think it would land us in a confusing situation. Can you take a look at the conversation I had with Mike in #testengineering yesterday and see if his proposed solution works for you?

Reply:
Read through the conversation, I think what you agreed on in chat makes sense to me.

The chat if others are curious: https://edx-internal.slack.com/archives/CCNL2NW5P/p1556826114026500